src/main/java/org/folio/fqm/service/MigrationService.java

```diff
@@ -64,4 +66,60 @@ public String getVersion(@CheckForNull String fqlQuery) {
       return migrationConfiguration.getDefaultVersion();
     }
   }
+
+  /**
+   * Migrates the query information and verifies that only the version has changed.
+   * If anything other than the version changes, throws a MigrationQueryChangedException.
+   *
+   * @param migratableQueryInformation the query information to migrate
+   * @throws MigrationQueryChangedException if anything other than the version changes
+   */
+  public void verifyQueryIsUpToDate(MigratableQueryInformation migratableQueryInformation) {
```

Comment:
I'd use a different name to make it clearer what this does (we have other `verify...` methods which check conditions and, if not matching the condition, update the data and return the updated version).

```suggestion
  public void throwExceptionIfQueryOutdated(MigratableQueryInformation migratableQueryInformation) {
```